pkg/cluster/cluster_config.go

```diff
@@ -53,16 +53,25 @@ func IsNotReservedNamespace(ns *corev1.Namespace) bool {
 // GetClusterServiceVersion retries the clusterserviceversions available in the operator namespace.
 func GetClusterServiceVersion(ctx context.Context, c client.Client, watchNameSpace string) (*ofapiv1alpha1.ClusterServiceVersion, error) {
 	clusterServiceVersionList := &ofapiv1alpha1.ClusterServiceVersionList{}
-	if err := c.List(ctx, clusterServiceVersionList, client.InNamespace(watchNameSpace)); err != nil {
-		return nil, fmt.Errorf("failed listing cluster service versions: %w", err)
+	listOptions := &client.ListOptions{
+		Limit:     100,
+		Namespace: watchNameSpace,
 	}
-
-	for _, csv := range clusterServiceVersionList.Items {
-		for _, operatorCR := range csv.Spec.CustomResourceDefinitions.Owned {
-			if operatorCR.Kind == "DataScienceCluster" {
-				return &csv, nil
+	for { // for the case we have very big size of CSV even just in one namespace
+		if err := c.List(ctx, clusterServiceVersionList, listOptions); err != nil {
+			return nil, fmt.Errorf("failed listing cluster service versions: %w", err)
+		}
+		for _, csv := range clusterServiceVersionList.Items {
+			for _, operatorCR := range csv.Spec.CustomResourceDefinitions.Owned {
+				if operatorCR.Kind == "DataScienceCluster" {
+					return &csv, nil
+				}
 			}
 		}
+		if clusterServiceVersionList.GetContinue() == "" {
+			break
+		}
+		listOptions.Continue = clusterServiceVersionList.GetContinue()
```

Comment:
Maybe we can make it one compact `if`. In a similar way to how we handle errors.

```suggestion
	if listOptions.Continue = clusterServiceVersionList.GetContinue(); listOptions.Continue == "" {
		break
	}
```

Reply:
interesting